durabletask/worker.py

```diff
@@ -88,15 +90,25 @@ def __init__(self, *,
                  metadata: Optional[list[tuple[str, str]]] = None,
                  log_handler=None,
                  log_formatter: Optional[logging.Formatter] = None,
-                 secure_channel: bool = False):
+                 secure_channel: bool = False,
+                 interceptors: Optional[list[grpc.ServerInterceptor]] = None):  # Add interceptors
         self._registry = _Registry()
         self._host_address = host_address if host_address else shared.get_default_host_address()
-        self._metadata = metadata
+        self._metadata = metadata or []  # Ensure metadata is never None
         self._logger = shared.get_logger("worker", log_handler, log_formatter)
         self._shutdown = Event()
         self._is_running = False
         self._secure_channel = secure_channel
 
+        # Determine the interceptors to use
+        if interceptors is not None:
+            self._interceptors = interceptors
+        elif self._metadata:
+            self._interceptors = [DefaultClientInterceptorImpl(self._metadata)]
```

Comment:
If someone provides both `interceptors` and `metadata`, then we probably want to do both the `DefaultClientInterceptorImpl` and also whatever interceptors that the caller provided, right? Otherwise we'd be totally ignoring the value they provided for `metadata`.